source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc

```diff
@@ -1,20 +1,81 @@
 #include "source/extensions/filters/http/ext_proc/http_client/http_client_impl.h"
 
 #include "source/common/common/enum_to_int.h"
+#include "source/common/http/header_map_impl.h"
+#include "source/common/http/message_impl.h"
 #include "source/common/http/utility.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace ExternalProcessing {
 
+void ExtProcHttpClient::sendRequest(envoy::service::ext_proc::v3::ProcessingRequest&& req) {
+  // Cancel any active requests.
+  cancel();
+
+  // Transcode req message into JSON string.
+  auto req_in_json = MessageUtil::getJsonStringFromMessage(req);
+  if (!req_in_json.ok()) {
+    ENVOY_LOG(error,
+              "The ProcessingRequest proto message can not be encoded into a JSON string: {}",
+              req_in_json.status().ToString());
+    onError();
+    return;
+  }
+
+  const std::string cluster = config_.http_service().http_service().http_uri().cluster();
+  const std::string url = config_.http_service().http_service().http_uri().uri();
```

Comment:
Is there any validation to make sure the URI is valid and produces valid host and path?

Reply:
This is Envoy filter configuration from trusted domain. I think we do not need validation.  If this turns out to be a bad configuration, the outcome will be no response received by ext_proc filter, and timeout happens. This will trigger customer to go back to fix the configuration.

Reply:
@yanavlasov I checked a couple of other Envoy call out extensions which use http_uri service, like ext_authz, no configuration validation there either. So, let's keep it this way.  Please let me know if you have concerns.